// back-end/prisma/constants/dataProducts.ts
import { uploadApi } from "./configs";

export const data = [
    {
        id: 1,
        cover: uploadApi+"/upload/products/bootstrap.png",
        title: "Bootstrap",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 150
    },
    {
        id: 2,
        cover: uploadApi+"/upload/products/javaScript.jpg",
        title: "Javascript",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 300
    },
    {
        id: 3,
        cover: uploadApi+"/upload/products/Full Stack Developer Career Program_Course Card copy.webp",
        title: "Full Stack",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 1500
    },
    {
        id: 4,
        cover: uploadApi+"/upload/products/MERN.jpg",
        title: "MERN",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 600
    },
    {
        id: 5,
        cover: uploadApi+"/upload/products/react-js.jpg",
        title: "React",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 300
    },
    {
        id: 6,
        cover: uploadApi+"/upload/products/html-css.jpg",
        title: "HTML&CSS",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 150
    },
    {
        id: 7,
        cover: uploadApi+"/upload/products/node.png",
        title: "Nodejs",
        description: "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Suscipit, dolorem, consequatur asperiores nulla error ratione unde culpa corrupti vel ipsa corporis possimus adipisci odio ipsam sequi nobis accusamus molestiae illo?",
        price: 300
    },
]
